### PR TITLE
de-frag translations

### LINKED ISSUE #2349: Fix hard-coded word order in translations.
From #1929. The outcome is to review the referenced code, see if the problem still exists, and if so resolve it.

------------------------------

In some languages, their grammar is vastly different from English, so if word order is fixed, translators cannot make natural translation.

For example:
screen

The order of these words is completely fixed in source code.
https://github.com/steemit/condenser/blob/master/src/app/components/cards/TransferHistoryRow.jsx

I want to translate this message "Transfer 0.200 SBD to inoue" into "inoueに0.200SBDを送りました" in Japanese.
However, because the word order is fixed, I translated it as "送信 0.200 SBD 宛先:inoue" in Japanese.
This is very unnatural.

Another one:

In left side of top page, if the user clicks "Show more", it will change to "Show fewer".

https://github.com/steemit/condenser/blob/master/src/app/components/pages/PostsIndex.jsx
line 177: <small><a className="c-sidebar__more-link" onClick={this.onShowSpam}>{showSpam ? tt('g.next_3_strings_together.show_less') : tt('g.next_3_strings_together.show_more')}</a>{' ' + tt('g.next_3_strings_together.value_posts')}</small>
The order of "Show more" (or "Show fewer") and "low value posts" can not be swapped.

I couldn't translate these into Japanese.
Because Japanese word order is SOV (subject-object-verb).
So I had to write "low value posts" blank and I wrote full text on "Show more" or "Show fewer".

### PR DESCRIPTION
**Pull request by bnchdrff**
closes #2349

fixes some of #2351 but that will need more work

still need to figure out a way to interpolate react dom elements into a translation (#2529)

also, this commit adds a storybook to cover all cases for TransferHistoryRow.

this story provides knobs for any incoming prop that affects branching and can be used as the basis for a real test eventually

### CODE REVIEW

File: src/app/components/cards/TransferHistoryRow/index.jsx [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+                type === 'transfer_to_savings'
+                    ? 'to_savings'
+                    : type === 'transfer_from_savings'
+                      ? 'from_savings'
+                      : 'not_savings';
+
+            if (data.from === context) {
+                // Semi-bad behavior - passing `type` to translation engine -- @todo better somehow?
+                // type can be to_savings, from_savings, or not_savings
+                // Also we can't pass React elements (link to other account) so its order is fixed :()
```
Comment (gl2748) [inline src/app/components/cards/TransferHistoryRow/index.jsx]: Bummer. For future reference what library supports passing components?

File: src/app/components/cards/TransferHistoryRow/story.jsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+import { receiveState } from 'app/redux/GlobalReducer';
+import mockState from '../../../../../api_mockdata/get_state';
+
+import TransferHistoryRow from './';
+
+const store = createStore(rootReducer);
+
+// Load some fake blockchain state.
+// This component relies on many parts of the state tree - messy & hard to test!
+store.dispatch(receiveState(mockState));
```
Comment (gl2748) [inline src/app/components/cards/TransferHistoryRow/story.jsx]: This is a good point of reference for working on other stories that will need store data.

**Comment by gl2748** [comment]
Hey @bnchdrff is there an order of work for updating the locales other than (en) to use the tokens now being passed?

**Comment by bnchdrff** [comment]
> Hey @bnchdrff is there an order of work for updating the locales other than (en) to use the tokens now being passed?

that's an @originated question!

File: src/app/components/cards/TransferHistoryRow/index.jsx (hunk 1 above)
Comment (bnchdrff) [reply src/app/components/cards/TransferHistoryRow/index.jsx]: https://github.com/martinandert/react-interpolate-component along with https://github.com/martinandert/react-translate-component is one option that is sorta workable with our existing system

https://react.i18next.com is another one

my fave so far is https://github.com/lingui/js-lingui

**Comment by gl2748** [comment]
@bnchdrff  - Getting missing translations warnings on the new storybook story for Transfer History Row.
Is this to be expected? Locale is set to 'EN' via the storybooks-intl knob-a-roo.
screenshot from 2018-02-20 09-11-51

**Comment by bnchdrff** [comment]
those errors are to be expected if i did a bad job filling in all of the new translations ;) fixing it now

**Comment by bnchdrff** [comment]
this needs more work; i forgot to put in the locale strings for the first component i did (TransferHistoryRow) and it'll take a minute; closing this until i fix that.

**Comment by bnchdrff** [comment]
@originated what should we do here? a bunch of new translations will be needed for this PR

**Comment by originated** [comment]
@bnchdrff Updates look good. To answer above currently flow is to make sure the en.json is always up to date in Transifix and have them translated first in the system. 

As far as getting the keys done promptly it's a product question for @pon if we should indeed be using a service, etc.

**Pull request merged**